flink-runtime/src/main/java/org/apache/flink/runtime/metrics/groups/AbstractMetricGroup.java

```diff
@@ -119,12 +122,28 @@ public AbstractMetricGroup(MetricRegistry registry, String[] scope, A parent) {
         this.variables = new Map[registry.getNumberReporters() + 1];
     }
 
+    CharacterFilter wrapWithDefaultFilter(CharacterFilter filter, char delimiter) {
+        CharacterFilter defaultFilter = input -> input.replace(delimiter, DEFAULT_REPLACEMENT);
+        return filter == null
+                ? defaultFilter
+                : input -> defaultFilter.filterCharacters(filter.filterCharacters(input));
+    }
+
     @Override
     public Map<String, String> getAllVariables() {
-        return internalGetAllVariables(0, Collections.emptySet());
+        return internalGetAllVariables(null, 0, Collections.emptySet(), registry.getDelimiter());
+    }
+
+    @Override
+    public Map<String, String> getAllVariables(CharacterFilter filter) {
+        return internalGetAllVariables(filter, 0, Collections.emptySet(), registry.getDelimiter());
     }
 
-    public Map<String, String> getAllVariables(int reporterIndex, Set<String> excludedVariables) {
+    public Map<String, String> getAllVariables(
+            CharacterFilter filter,
+            int reporterIndex,
+            Set<String> excludedVariables,
+            char delimiter) {
```

Comment:
I'd like to keep the reporterIndex as the first argument because all other arguments are somewhat dependent on it

Reply:
+1